convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,205 @@
+            try:
+                shutil.copy2(src_path, dst_path)
+            except (OSError, IOError) as err:
+                # IOError for py2 and OSError for py3
+                self.set_result(
+                    level="ERROR",
+                    id="IO_ERROR",
+                    title="I/O error",
+                    description=(
+                        "I/O error(%s): %s Some GRUB files have not been copied to /boot/efi/EFI/redhat."
+                        % (err.errno, err.strerror)
+                    ),
+                )
+
+
+class RemoveEfiCentos(actions.Action):
+    id = "REMOVE_EFI_CENTOS"
+    dependencies = ("COPY_GRUB_FILES",)
+
+    def run(self):
+        """Remove the /boot/efi/EFI/centos/ directory when no UEFI files remains.
+
+        The centos/ directory after the conversion contains usually just grubenv,
+        grub.cfg, .. files only. Which we copy into the redhat/ directory. If no
+        other UEFI files are present, we can remove this dir. However, if additional
+        UEFI files are present, we should keep the directory for now, until we
+        deal with it.
+        """
+        super(RemoveEfiCentos, self).run()
+
+        if systeminfo.system_info.id != "centos":
+            logger.debug("Skipping removing EFI files - only related to CentOS Linux.")
+            # nothing to do
+            return
+        try:
+            os.rmdir(CENTOS_EFIDIR_CANONICAL_PATH)
+        except OSError:
+            warning_message = (
+                "The folder %s is left untouched. You may remove the folder manually"
+                " after you ensure there is no custom data you would need." % CENTOS_EFIDIR_CANONICAL_PATH
+            )
+            logger.warning(warning_message)
+            self.add_message(
+                level="WARNING",
+                id="FOLDER_NOT_REMOVED",
+                title="Folder was not removed",
+                description=warning_message,
+            )
+
+
+class ReplaceEfiBootEntry(actions.Action):
+    id = "REPLACE_EFI_BOOT_ENTRY"
+    dependencies = ("REMOVE_EFI_CENTOS",)
+
+    def run(self):
+        """Replace the current UEFI bootloader entry with the RHEL one.
+
+        The current UEFI bootloader entry could be invalid or misleading. It's
+        expected that the new bootloader entry will refer to one of the standard UEFI binary
+        files provided by Red Hat inside the RHEL_EFIDIR_CANONICAL_PATH.
+        The new UEFI bootloader entry is always created / registered and set
+        set as default.
+
+        The current (original) UEFI bootloader entry is removed under some conditions
+        (see _remove_orig_boot_entry() for more info).
+        """
+        super(ReplaceEfiBootEntry, self).run()
+
+        try:
+            grub.replace_efi_boot_entry()
+        except grub.BootloaderError as e:
+            self.set_result(
+                level="ERROR",
+                id="BOOTLOADER_ERROR",
+                title="Bootloader error",
```

Comment:
Can we get a more descriptive title? Something like `Failed to replace EFI boot entry.`

If we can improve the `id` as well, would be good.

Reply:
I think this need some improvements still, @pr-watson

Reply:
Sorry I thought I updated this, correcting it now